packages/notifications/src/notificationsMiddleware/notificationsMiddleware.ts

```diff
@@ -50,16 +77,16 @@ export const createNotificationsMiddleware = (options = {}) => {
     };
     const middlewareOptions = { ...defaultOptions, ...options };
 
-    const matchPending = type => type.match(new RegExp(`^.*${middlewareOptions.pendingSuffix}$`));
-    const matchFulfilled = type => type.match(new RegExp(`^.*${middlewareOptions.fulfilledSuffix}$`));
-    const matchRejected = type => type.match(new RegExp(`^.*${middlewareOptions.rejectedSuffix}$`));
+    const matchPending = (type: string) => Boolean(type.match(new RegExp(`^.*${middlewareOptions.pendingSuffix}$`)));
+    const matchFulfilled = (type: string) => Boolean(type.match(new RegExp(`^.*${middlewareOptions.fulfilledSuffix}$`)));
+    const matchRejected = (type: string) => Boolean(type.match(new RegExp(`^.*${middlewareOptions.rejectedSuffix}$`)));
```

Comment:
Might be cool to use [user-defined-type-guards](https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards), it would require a bit of rewrite, so I'll leave it up to you (you can open a new issue to address this later)

Reply:
I would do this later on. Same as with the class. I don't want to bother with that right now. I just need something published to verify the build and package works.

Reply:
Good point! Let's create an issue and implement it later on.